### PR TITLE
Use Snyk orb

### PR DESCRIPTION
**Pull request by rmorais**

### CODE REVIEW

File: rac-gcp-deploy/orb.yml [hunk 1]
```diff
@@ -288,6 +281,88 @@ jobs:
...
+    executor: <<parameters.executor>>
+    steps:
+      - attach_workspace:
+          at: <<parameters.workspace-dir>>
+      - load_cache:
+          cache-suffix: <<parameters.cache-suffix>>
+      - run: *sbt-dependency-graph
+      - snyk/scan:
+          project: $CIRCLE_PROJECT_REPONAME
+          fail-on-issues: true
```
Comment (jensraaby) [inline rac-gcp-deploy/orb.yml:307]: Will there be a way to override the fail-on-issues if we get stuck with a particular dependency?
(It's a bit of a risk though as you'd be allowing broken windows...)

Comment (rmorais) [reply rac-gcp-deploy/orb.yml:307]: At the moment there isn't any override.

File: rac-gcp-deploy/orb.yml [hunk 2]
```diff
@@ -288,6 +281,115 @@ jobs:
...
+          at: <<parameters.workspace-dir>>
+      - load_cache:
+          cache-suffix: <<parameters.cache-suffix>>
+      - run: *sbt-dependency-graph
+      - snyk/scan:
+          project: $CIRCLE_PROJECT_REPONAME
+          fail-on-issues: false
+          monitor-on-build: true
+
+  pr_comment_snyk:
```
Comment (redbaron) [inline rac-gcp-deploy/orb.yml:337]: IMHO these should be commands, not jobs. Jobs require start of the new executor which increases contention on CircleCI and makes builds longer

File: rac-gcp-deploy/orb.yml [hunk 3]
```diff
@@ -288,6 +281,115 @@ jobs:
...
+      - load_cache:
+          cache-suffix: <<parameters.cache-suffix>>
+      - run: *sbt-dependency-graph
+      - run:
+          name: Post Snyk report to GitHub PR
+          command: |
+            PR_COMMENT_TITLE="SNYK REPORT"
+
+            comment_pr () {
+              local pr_body="$PR_COMMENT_TITLE"$'\n```PowerShell\n'"$snyk_report"$'\n```'
```
Comment (redbaron) [inline rac-gcp-deploy/orb.yml:364]: what is `$` in `$'\n` ?

Comment (rmorais) [reply rac-gcp-deploy/orb.yml:364]: some bash syntax to allow the correct handling of `\n` in strings

File: rac-gcp-deploy/orb.yml [hunk 4]
```diff
@@ -288,6 +281,115 @@ jobs:
...
+    executor: <<parameters.executor>>
+    steps:
+      - attach_workspace:
+          at: <<parameters.workspace-dir>>
+      - load_cache:
+          cache-suffix: <<parameters.cache-suffix>>
+      - run: *sbt-dependency-graph
+      - run:
+          name: Post Snyk report to GitHub PR
+          command: |
```
Comment (tcsquires) [inline rac-gcp-deploy/orb.yml:360]: Can this (should this) Bash be extracted into a `.sh` script file that the Orb invokes?

File: rac-gcp-deploy/orb.yml (hunk 2 above)
Comment (rmorais) [reply rac-gcp-deploy/orb.yml:337]: By these, you are referring to all of the snyk jobs or just the `pr_comment_snyk`one?

File: rac-gcp-deploy/orb.yml (hunk 3 above)
Comment (redbaron) [reply rac-gcp-deploy/orb.yml:364]: TIL

File: rac-gcp-deploy/orb.yml (hunk 4 above)
Comment (rmorais) [reply rac-gcp-deploy/orb.yml:360]: I think it can. I will have a look at it

File: rac-gcp-deploy/orb.yml (hunk 2 above)
Comment (rmorais) [reply rac-gcp-deploy/orb.yml:337]: I can make the snyk_monitor a job and include it as a step of the release job.
As for the other two jobs, I don't think they be made simple commands and included as a step on the existing jobs. These should be run only for PR's but we don't any specific jobs only for PRs.
I happy to hear any suggestions you might have.

File: rac-gcp-deploy/orb.yml (hunk 4 above)
Comment (redbaron) [reply rac-gcp-deploy/orb.yml:342]: not possible AFAIK :( only option is inline orb or bake it into an docker image orb must be used with

File: rac-gcp-deploy/orb.yml [hunk 5]
```diff
@@ -340,37 +340,7 @@ jobs:
...
-              -u $GITHUB_USERNAME:$GITHUB_TOKEN | jq -r '.[] | select(.user.login == "'$GITHUB_USERNAME'") | select(.body | startswith("'"$PR_COMMENT_TITLE"'")) | .url')
-
-            if [ -z "$existing_pr_comment" ]; then
-              echo "Adding new comment"
-              comment_pr $pr_comment_url
-            else
-              echo "Updating existing comment"
-              comment_pr $existing_pr_comment
-            fi
+            include github_comment_pr.sh
```
Comment (redbaron) [inline rac-gcp-deploy/orb.yml:343]: what is this magic?

Comment (rmorais) [reply rac-gcp-deploy/orb.yml:343]: not sure if it works but other orbs in this repo, inline scripts like this.
If and when CircleCi is functioning properly again, I will tell you if it works

Comment (redbaron) [reply rac-gcp-deploy/orb.yml:343]: OMG, orbs are preprocessed before being published: https://github.com/ovotech/circleci-orbs/blob/7ef7e85c8d505f2ff01604483e8f11f0d9a9c8aa/tools/include.py

Comment (rmorais) [reply rac-gcp-deploy/orb.yml:343]: so this actually seems to work :)
the external script was inlined in the compiled configuration

Comment (rmorais) [reply rac-gcp-deploy/orb.yml:343]: I guess that preprocessor is doing the same as https://circleci.com/docs/2.0/using-orbs/#file-include-syntax

Comment (redbaron) [reply rac-gcp-deploy/orb.yml:343]: Nice find!

**Review by redbaron — APPROVED** [review]

**Review by sophiecollard — APPROVED** [review]
Approving based on your demo of this 😄

**Pull request merged**